### PR TITLE
Fix typos and rendering issues in docs

### PR DESCRIPTION
**Pull request by eahlberg**
Fixes a couple of typos and Haddock rendering issues.

Before submitting your PR, check that you've:
- [x] Ran `stylish-haskell` on any changed files.
- [x] Adhered to the code style (see the `.editorconfig` file for details)

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

File: persistent/Database/Persist/Quasi.hs [hunk 1]
```diff
@@ -203,7 +203,7 @@ You can automatically get ToJSON and FromJSON instances for any entity by adding
 Person json
     name Text
 @
-Requires @{-# LANGUAGE FlexibleInstances #-}@
+Requires @\{\-\# LANGUAGE FlexibleInstances \#\-\}@
```
Comment (eahlberg) [inline persistent/Database/Persist/Quasi.hs:206]: `Requires @{-# LANGUAGE FlexibleInstances #-}@` would render as `@@`, see under (JSON instances)[https://hackage.haskell.org/package/persistent-2.13.1.2/docs/Database-Persist-Quasi.html]

File: persistent/Database/Persist.hs [hunk 2]
```diff
@@ -16,7 +16,7 @@ module Database.Persist
 -- This syntax allows you to customize the resulting Haskell datatypes and
 -- database schema. See "Database.Persist.Quasi" for details on that definition
 -- language.
---
+
```
Comment (eahlberg) [inline persistent/Database/Persist.hs:19]: This was needed to properly create a heading for `Reference Schema & Dataset` [here](https://hackage.haskell.org/package/persistent-2.13.1.2/docs/Database-Persist.html)

File: persistent/Database/Persist/Quasi.hs [hunk 3]
```diff
@@ -216,7 +216,7 @@ Person sql=peoples
     name Text
 @
 
-== Change table/collection key definition (field name and/or type, persistent >= 2.1)
+== Change table/collection key definition (field name and\/or type, persistent >= 2.1)
```
Comment (eahlberg) [inline persistent/Database/Persist/Quasi.hs:219]: Escaping of the 2nd `/` was needed to render properly, see [this page](Change tablecollection key definition (field name andor type, persistent >= 2.1).

**Comment by eahlberg** [comment]
Found a couple of small issues in the docs that I figured I could have a go at. Since I'm a first-time contributor, CI won't run automatically so I haven't been able to see if it succeeds and cross off the task above, but the PR should be ready for review!

**Review by parsonsmatt — APPROVED** [review]
This is fantastic! Thank you 😄

**Comment by eahlberg** [comment]
No worries!

**Pull request merged**